### PR TITLE
OCaml scripting demo!

### PR DESCRIPTION
**Pull request by aryx**
test plan:
```
$ opam install -y dev.opam
$ ./scripts/hello_script.ml --verbose world
debug: params = { Hello_script.username = "pad"; verbose = true; command = "world" }
Hello world from pad
PAD      1871921  0.0  0.0 194628 32096 PTS/0    SL+  17:07   0:00 /HOME/PAD/.OPAM/4.14.0/BIN/OCAMLRUN /HOME/PAD/.OPAM/4.14.0/BIN/OCAML ./SCRIPTS/HELLO_SCRIPT.ML --VERBOSE WORLD
PAD      1871940  0.0  0.0   6512  2560 PTS/0    S+   17:07   0:00 GREP OCAML
```

PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
This is a followup to some discussions on https://github.com/returntocorp/semgrep/pull/6300

File: scripts/hello_script.ml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+(* You need first to 'opam install ppx_deriving'. *)
+#require "ppx_deriving.show"
+
+(* You need first to 'opam install ppx_deriving_cmdliner'. *)
+#require "ppx_deriving_cmdliner"
+
+#load "unix.cma"
+
+#load "threads/threads.cma"
```
Comment (mjambon) [inline scripts/hello_script.ml:48]: You can (should) use:
```
#require "unix"
#require "threads"
```

Comment (mjambon) [reply scripts/hello_script.ml:48]: Wait, we don't use threads. At least not directly. Do we really need `#require "threads"`?

File: scripts/hello_script.ml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+(*****************************************************************************)
+(* Cmdliner boilerplate *)
+(*****************************************************************************)
+let main () =
+  let info = Cmd.info Sys.argv.(0) in
+  let term = Term.(const run $ cli_params_cmdliner_term ()) in
+  let cmd = Cmd.v info term in
+  exit (Cmd.eval cmd)
+
+let () = main ()
```
Comment (mjambon) [inline scripts/hello_script.ml:113]: We should probably create a library function that takes care of the cmdliner incantations. Something like
```ocaml
(* Take a command-line parser, execute a program, and exit. *)
val main : 'a Cmdliner.Term.t -> ('a -> int) -> 'b
```

Comment (mjambon) [reply scripts/hello_script.ml:113]: Such a `main` utility should also provide stack backtrace recording. Maybe also a few other things I'm not thinking of right now.

File: scripts/hello_script.ml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+(* see https://github.com/charlesetc/feather *)
+let simple_shell_programming_demo () =
+  let out =
+    F.process "ps" [ "-aux" ]
+    |. F.map_lines ~f:String.uppercase_ascii
+    |. F.process "grep" [ "OCAML" ]
+    (* alt: |. F.grep "OCAML" *)
+    |> F.collect F.stdout
+  in
+  pr out
```
Comment (mjambon) [inline scripts/hello_script.ml:92]: not bad!

**Pull request merged**